### PR TITLE
Replace typedefs with using

### PR DESCRIPTION
**Pull request by aurianer**
I thought of doing that in one go instead of piece by piece
EDIT: my scripts missed some, I have to figure out a way to catch those ^^'

### CODE REVIEW

File: examples/quickstart/sort_by_key_demo.cpp [hunk 1]
```diff
@@ -51,6 +51,5 @@ int pika_main()
 ///////////////////////////////////////////////////////////////////////////////
 int main(int argc, char* argv[])
 {
-    return pika::init(
-        pika_main, argc, argv);    // Initialize and run pika
+    return pika::init(pika_main, argc, argv);    // Initialize and run pika
```
Comment (msimberg) [inline examples/quickstart/sort_by_key_demo.cpp:54]: Unrelated to this PR, but we can probably have the clang-format check enabled for all source files now (including .cu files), right?

**Review by msimberg — COMMENTED** [review]
Excellent idea. Could you try enabling https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-using.html in our clang-tidy config after this as well?

File: examples/quickstart/sort_by_key_demo.cpp (hunk 1 above)
Comment (msimberg) [reply examples/quickstart/sort_by_key_demo.cpp:54]: And related to clang-format, I wouldn't mind bumping the clang version if needed.

**Comment by aurianer** [comment]
Is that okay if I look into the rest of the typedefs in another PR? Cause it conflicts quite much with the cleanup PRs
Otherwise any idea about the codespell issue? Is it possible to ignore that file or that url in particular?

**Comment by aurianer** [comment]
> Excellent idea. Could you try enabling https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-using.html in our clang-tidy config after this as well?

I opened #169 to remember that

File: examples/quickstart/sort_by_key_demo.cpp (hunk 1 above)
Comment (aurianer) [reply examples/quickstart/sort_by_key_demo.cpp]: I opened #168

**Comment by msimberg** [comment]
> Is that okay if I look into the rest of the typedefs in another PR? Cause it conflicts quite much with the cleanup PRs

Absolutely.

> Otherwise any idea about the codespell issue? Is it possible to ignore that file or that url in particular?

I wouldn't want to ignore the whole file, and the check is not required. As far as I know it's only possible to have either a global whitelist of spellings, or ignore files, but not possible to have a whitelist for a specific file/line. I think we can just ignore it.

**Review by msimberg — APPROVED** [review]

**Comment by msimberg** [comment]
bors merge

**Pull request merged**